### ISSUE TITLE
Upgrading syncthing-relaysrv to 1.15.0 fails in post-installation

### ISSUE THREAD

**Issue by bronikowski** [syncthing/syncthing#7548]
* Versions: Upgrading the to the latest release (from candidate to stable) on ARM
* What happened: The dist-upgrade fails with an error in a post-installation script
* What was expected: your run-of-the-mill upgrade

Output:

```
Setting up syncthing-relaysrv (1.15.0) ...
/var/lib/dpkg/info/syncthing-relaysrv.postinst: 11: /var/lib/dpkg/info/syncthing-relaysrv.postinst: Syntax error: "else" unexpected
dpkg: error processing package syncthing-relaysrv (--configure):
 installed syncthing-relaysrv package post-installation script subprocess returned error exit status 2
Errors were encountered while processing:
 syncthing-relaysrv
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Comment by calmh** [comment]
Far as I can tell we don't actually include a postinst script for the relay server. What does that script look like?

**Comment by bronikowski** [comment]
Sure, here's what I get when I `less` the file. 

```
#!/bin/sh


after_upgrade() {
    :

systemctl --system daemon-reload >/dev/null || true
debsystemctl=$(command -v deb-systemd-invoke || echo systemctl)
if ! systemctl is-enabled strelaysrv >/dev/null
then
else
    $debsystemctl restart strelaysrv >/dev/null || true
fi
}

after_install() {
    :

systemctl --system daemon-reload >/dev/null || true
debsystemctl=$(command -v deb-systemd-invoke || echo systemctl)
}

if [ "${1}" = "configure" -a -z "${2}" ] || \
   [ "${1}" = "abort-remove" ]
then
    # "after install" here
    # "abort-remove" happens when the pre-removal script failed.
    #   In that case, this script, which should be idemptoent, is run
    #   to ensure a clean roll-back of the removal.
    after_install
elif [ "${1}" = "configure" -a -n "${2}" ]
then
    upgradeFromVersion="${2}"
    # "after upgrade" here
    # NOTE: This slot is also used when deb packages are removed,
    # but their config files aren't, but a newer version of the
    # package is installed later, called "Config-Files" state.
    # basically, that still looks a _lot_ like an upgrade to me.
    after_upgrade "${2}"
elif echo "${1}" | grep -E -q "(abort|fail)"
then
    echo "Failed to install before the post-installation script was run." >&2
    exit 1
fi```

**Comment by calmh** [comment]
Apparently this is something auto generated by the packaging script because we say we have a systemd service. 🤷

**Comment by bronikowski** [comment]
Ah, the wonders of packaging. Clearly the `if` is there to prod the running process to `HUP`.

**Comment by imsodin** [comment]
https://github.com/jordansissel/fpm/issues/1757

**Comment by js-d-coder** [comment]
I fixed this installation issue by adding `echo` after line 10 in `/var/lib/dpkg/info/syncthing-relaysrv.postinst` and then ran `sudo apt-get update && sudo apt-get upgrade && sudo apt-get dist-upgrade && sudo apt-get install -f`. Voila, it worked. ✌🏻

**Comment by neckcen** [comment]
Good day,

I'd like to point out that as of today, using official syncthing repositories, the last available package for syncthing-relaysrv is still broken. Whilst it is fairly easy to work around the bug, could you consider releasing an upgraded package with the fix included?

```
# apt policy syncthing-relaysrv
syncthing-relaysrv:
  Installed: (none)
  Candidate: 1.15.0
  Version table:
     1.15.0 500
        500 http://apt.syncthing.net syncthing/release amd64 Packages
```

Thank you for your work.

**Comment by calmh** [comment]
There should be 1.18.3 ones out now, which I'm hoping are fixed.
